pkg/api/handlers/compat/images.go

```diff
@@ -404,26 +405,44 @@ func GetImage(w http.ResponseWriter, r *http.Request) {
 }
 
 func GetImages(w http.ResponseWriter, r *http.Request) {
-	images, err := utils.GetImages(w, r)
+	decoder := r.Context().Value(api.DecoderKey).(*schema.Decoder)
+	runtime := r.Context().Value(api.RuntimeKey).(*libpod.Runtime)
+	query := struct {
+		All     bool
+		Digests bool
+		Filter  string // Docker 1.24 compatibility
+	}{
+		// This is where you can override the golang default value for one of fields
+	}
+
+	if err := decoder.Decode(&query, r.URL.Query()); err != nil {
+		utils.Error(w, http.StatusText(http.StatusBadRequest), http.StatusBadRequest,
+			errors.Wrapf(err, "failed to parse parameters for %s", r.URL.String()))
+		return
+	}
+	if _, found := r.URL.Query()["digests"]; found && query.Digests {
+		utils.UnSupportedParameter("digests")
+		return
+	}
+
+	filterList, err := filters.FiltersFromRequest(r)
 	if err != nil {
-		utils.Error(w, "Something went wrong.", http.StatusInternalServerError, errors.Wrap(err, "Failed get images"))
+		utils.Error(w, "Something went wrong.", http.StatusInternalServerError, err)
 		return
 	}
+	if !utils.IsLibpodRequest(r) && len(query.Filter) > 0 { // Docker 1.24 compatibility
+		filterList = append(filterList, "reference="+query.Filter)
+	}
 
-	summaries := make([]*entities.ImageSummary, 0, len(images))
-	for _, img := range images {
-		// If the image is a manifest list, extract as much as we can.
-		if isML, _ := img.IsManifestList(r.Context()); isML {
```

Comment:
For `compat` api we were ignoring `manifest` list I think merging this with common code would start printing `manifest` lists as well. 

I have only looked at the code yet. But I'll need to try this out inoder to make it sure that we are not breaking any existing behavior.

Reply:
This filter
+               filterList = append(filterList, "manifest=false")

Should remove all manifest images for the DockerAPI.